HDIClusterConfigurationManagement/src/main/java/com/microsoft/configuration/management/Main.java

```diff
@@ -230,21 +294,23 @@ public static void UploadConfiguration(URL targetAmbariHost, String targetAmbari
                         desiredConfig.type = downloadComponent.getKey();
                         desiredConfig.tag = (tag == null) ? (downloadComponent.getKey() + Long.toString(System.currentTimeMillis())) : tag.getAsString();
                         desiredConfig.properties = downloadedComponentConfig.items.get(0).properties;
+                        desiredConfig.properties_attributes = propertiesAttributes;
+                        desiredConfig.service_config_version_note = "Configs updated by HDInsight Cluster Configuration Management client tool.";
                         ClusterConfig config = new ClusterConfig();
                         config.desired_config = desiredConfig;
                         NewClusterConfig newClusterConfig = new NewClusterConfig();
                         newClusterConfig.Clusters = config;
 
                         result = SetNewConfiguration(targetAmbariHost, targetAmbariUsername, targetAmbariPassword, targetClusterName, newClusterConfig);
                         if (result.ResponseCode != 200) {
-                            // throw exception
+                            throw new RuntimeException("Failed to update configuration for " + downloadComponent.getKey());
                         }
                         configurationUpdated = true;
                     }
                 }
             }
 
-            if (configurationUpdated) {
+            if (configurationUpdated && shouldRestartService) {
```

Comment:
If shoudRestartService is false, we can show the list of services that have to be restarted so the configurations will take effect.